src/Language/PureScript/Ide/Imports.hs

```diff
@@ -0,0 +1,193 @@
+-----------------------------------------------------------------------------
+--
+-- Module      : Language.PureScript.Ide.Imports
+-- Description : Provides functionality to manage imports
+-- Copyright   : Christoph Hegemann 2016
+-- License     : MIT (http://opensource.org/licenses/MIT)
+--
+-- Maintainer  : Christoph Hegemann <christoph.hegemann1337@gmail.com>
+-- Stability   : experimental
+--
+-- |
+-- Provides functionality to manage imports
+-----------------------------------------------------------------------------
+
+{-# LANGUAGE ConstraintKinds       #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE PackageImports        #-}
+
+module Language.PureScript.Ide.Imports
+       ( addImplicitImport
+       , addImportForIdentifier
+       , answerRequest
+         -- for tests
+       , addImplicitImport'
+       , addExplicitImport'
+       , sliceImportSection
+       , prettyPrintImport'
+       , Import(Import)
+       )
+       where
+
+import           Control.Monad.Error.Class
+import           Control.Monad.IO.Class
+import           "monad-logger" Control.Monad.Logger
+import           Data.Function (on)
+import qualified Data.List                          as List
+import           Data.Maybe                         (mapMaybe)
+import           Data.Monoid                        ((<>))
+import           Data.Text                          (Text)
+import qualified Data.Text                          as T
+import qualified Data.Text.IO                       as TIO
+import qualified Language.PureScript                as P
+import           Language.PureScript.Ide.Completion
+import           Language.PureScript.Ide.Error
+import           Language.PureScript.Ide.Filter
+import           Language.PureScript.Ide.State
+import           Language.PureScript.Ide.Types
+
+data Import = Import P.ModuleName P.ImportDeclarationType  (Maybe P.ModuleName)
+              deriving (Eq, Show)
+
+-- | Reads a file and returns the (lines before the imports, the imports, the
+-- lines after the imports)
+parseImportsFromFile :: (MonadIO m, MonadError PscIdeError m) =>
+                        FilePath -> m ([Text], [Import], [Text])
+parseImportsFromFile fp = do
+  file <- liftIO (TIO.readFile fp)
+  pure (sliceImportSection (T.lines file))
+
+sliceImportSection :: [Text] -> ([Text], [Import], [Text])
+sliceImportSection ls =
+  let
+    preImportSection = takeWhile (not . hasImportPrefix) ls
+    importSection =
+      takeWhile continuesImport $
+        dropWhile (not . hasImportPrefix) ls
+    postImportSection =
+      dropWhile continuesImport $
+        dropWhile (not . hasImportPrefix) ls
+    hasImportPrefix = T.isPrefixOf "import"
+    continuesImport x = hasImportPrefix x || T.isPrefixOf " " x || x == ""
```

Comment:
What is the `isPrefixOf " "` option for here?

How would you handle imports preceded by whitespace? Or punctuated with comments?

Why not just parse the whole module here?